python/cudf/cudf/utils/utils.py

```diff
@@ -70,7 +70,14 @@ def scalar_broadcast_to(scalar, size, dtype=None):
         return column_empty(size, dtype=dtype, masked=True)
 
     if isinstance(scalar, pd.Categorical):
-        return scalar_broadcast_to(scalar.categories[0], size).astype(dtype)
+        if dtype is None:
+            return scalar_broadcast_to(scalar.categories[0], size).astype(
+                scalar.dtype
+            )
```

Comment:
This feels crazy inefficient. We pull out the say string scalar here, broadcast that, then recast it back to a category, instead of just broadcasting the integer underneath it and reusing the category.

Reply:
So this is only called for a single row frames so i am not to worried about efficiency here.   

https://github.com/rapidsai/cudf/blob/2d79ef57864eb25012408b7957a06c1e029f443a/python/cudf/cudf/utils/dtypes.py#L144-L154

I can switch it out to broadcast underlying categories if thats more preferable . 

https://github.com/rapidsai/cudf/issues/2269#issuecomment-634936509

Reply:
`scalar` is only a single row, but `size` can be arbitrary. Based on the current implementation say the on category was `abcdefghijklmnopqrstuvwxyz`, we end up broadcasting that huge string to `size` and then dictionary encoding it, instead of broadcasting `0` to `size` and then applying the dtype with the dictionary.

Reply:
Yup, that makes sense, will switch the implementation. 

Sorry for the back and forth here.